example/lib/test_scenarios/sensor_tag_test_scenario.dart

```diff
@@ -1,97 +1,195 @@
 part of test_scenarios;
 
-class SensorTagTestScenario implements TestScenario {
-  BleManager bleManager = BleManager.getInstance();
-  bool deviceConnectionAttempted = false;
+class SensorTagTestScenario extends TestScenario {
+  final Peripheral _peripheral;
 
+  SensorTagTestScenario(this._peripheral);
+
+  @override
```

Comment:
what about pass loggers as the constructor parameters? then we wouldn't have to pass it to every private methods?